lib/remap-functions/src/parse_common_log.rs

```diff
@@ -0,0 +1,242 @@
+    // - W3C specification: https://www.w3.org/Daemon/User/Config/Logging.html#common-logfile-format
+    // - Apache HTTP Server docs: https://httpd.apache.org/docs/1.3/logs.html#common
+    static ref REGEX_COMMON_LOG: Regex = Regex::new(
+        r#"(?x)                                 # Ignore whitespace and comments in the regex expression.
+        ^\s*                                    # Start with any number of whitespaces.
+        (-|(?P<host>.*?))\s+                    # Match `-` or any character (non-greedily) and at least one whitespace.
+        (-|(?P<identity>.*?))\s+                # Match `-` or any character (non-greedily) and at least one whitespace.
+        (-|(?P<user>.*?))\s+                    # Match `-` or any character (non-greedily) and at least one whitespace.
+        (-|\[(-|(?P<timestamp>[^\[]*))\])\s+    # Match `-` or `[` followed by `-` or any character except `]`, `]` and at least one whitespace.
+        (-|"(-|(\s*                             # Match `-` or `"` followed by `-` or and any number of whitespaces...
+        (?P<message>(                           # Match a request with...
+        (?P<method>\w+)\s+                      # Match at least one word character and at least one whitespace.
+        (?P<path>[[\\"][^"]]*?)\s+              # Match any character except `"`, but `\"` (non-greedily) and at least one whitespace.
+        (?P<protocol>[[\\"][^"]]*?)\s*          # Match any character except `"`, but `\"` (non-greedily) and any number of whitespaces.
+        |[[\\"][^"]]*?))\s*))"                  # ...Or match any charater except `"`, but `\"`, and any amount of whitespaces.
+        )\s+                                    # Match at least one whitespace.
+        (-|(?P<status>\d+))\s+                  # Match `-` or at least one digit and at least one whitespace.
+        (-|(?P<size>\d+))                       # Match `-` or at least one digit.
+        \s*                                     # Match and any number of whitespaces.
+    "#)
+    .expect("failed compiling regex for common log");
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseCommonLog;
+
+impl Function for ParseCommonLog {
+    fn identifier(&self) -> &'static str {
+        "parse_common_log"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "timestamp_format",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+        let timestamp_format = arguments.optional_literal("timestamp_format")?.map_or(
```

Comment:
I don't think this needs to be a literal string. Generally parameters only need to be literals if there is a performance cost to evaluating it - eg. regular expressions, the grok pattern. Everything else can be evaluated at run time. Code such as the following could be useful to support:

```coffeescript
if .country = "uk" {
  format = "%d/%m/%y"
} else {
  format = "%m/%d/%y"
}

parse_common_log(..., format)
```